inst/examples/germany_latent_renewal.R

```diff
@@ -0,0 +1,156 @@
+# Load packages
+library(epinowcast)
+library(data.table)
+library(ggplot2)
+
+# Use 2 cores
+options(mc.cores = 2)
+
+# Load and filter germany hospitalisations
+nat_germany_hosp <- germany_covid19_hosp[location == "DE"][age_group %in% "00+"]
+nat_germany_hosp <- enw_filter_report_dates(
+  nat_germany_hosp,
+  latest_date = "2021-10-01"
+)
+
+# Make sure observations are complete
+nat_germany_hosp <- enw_complete_dates(
+  nat_germany_hosp,
+  by = c("location", "age_group")
+)
+# Make a retrospective dataset
+retro_nat_germany <- enw_filter_report_dates(
+  nat_germany_hosp,
+  remove_days = 40
+)
+retro_nat_germany <- enw_filter_reference_dates(
+  retro_nat_germany,
+  include_days = 40
+)
+
+# Get latest observations for the same time period
+latest_obs <- enw_latest_data(nat_germany_hosp)
+latest_obs <- enw_filter_reference_dates(
+  latest_obs,
+  remove_days = 40, include_days = 20
+)
+
+# Preprocess observations (note this maximum delay is likely too short)
+pobs <- enw_preprocess_data(retro_nat_germany, max_delay = 20)
+
+# Compile nowcasting model using multi-threading
+model <- enw_model(threads = TRUE, stanc_options = list("O1"))
+
+# Reference date model: Fixed log-normal distribution
+reference_module <- enw_reference(~1, data = pobs)
+
+# Report date model: Day of week reporting effect
+report_module <- enw_report(~ (1 | day_of_week), data = pobs)
+
+# Expectation model:
+# - Weekly random walk on the growth rate
+# - Generation time with probability mass spread over 4 days
+# - latent reporting delay representing the incubation period and assumed
+# ascertainment
+# - Day of week reporting effect
+expectation_module <- enw_expectation(
+  r = ~ rw(week),
+  generation_time = c(0.1, 0.4, 0.4, 0.1),
+  observation = ~ (1 | day_of_week),
+  latent_reporting_delay = 0.4 * c(0.05, 0.3, 0.6, 0.05),
+  data = pobs
+)
```

Comment:
When testing this, I get a warning message

> Warning message:
In enw_add_metaobs_features(r_features, ...) :
  Pre-existing columns in `metaobs` will be overwritten: {day_of_week}.Pre-existing columns in `metaobs` will be overwritten: {day}.Pre-existing columns in `metaobs` will be overwritten: {week}.Pre-existing columns in `metaobs` will be overwritten: {month}.

Reply:
This is from a recent PR from @pearsonca and isn't obviously desirable. We basically have to recreate the date helper variables as we have extended the date range of the data. In the current setup this throws a warning. We likely want to suppress this and ideally think of a more elegant way to handle this (though not clear what that is).

Reply:
I've suppressed warnings from this function call for now.